### PR TITLE
ENH: add sort_categories argument to union_categoricals

### PR DESCRIPTION
**Pull request by chris-b1**
- needed for #13406, follow-up to #13763
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

cc: @janschulz, @sinhrks

### CODE REVIEW

File: pandas/types/concat.py [hunk 1]
```diff
@@ -244,41 +245,47 @@ def union_categoricals(to_union):
...
-        return Categorical(np.concatenate([c.codes for c in to_union]),
-                           categories=first.categories, ordered=first.ordered,
-                           fastpath=True)
+        # identical categories - fastpath
+        categories = first.categories
+        ordered = first.ordered
+        new_codes = np.concatenate([c.codes for c in to_union])
+
+        if sort_categories:
+            categories = categories.sort_values()
```
Comment (sinhrks) [inline pandas/types/concat.py]: I think sort can be skipped if categories is monotonic-increasing.

File: pandas/tools/tests/test_concat.py [hunk 2]
```diff
@@ -989,6 +989,50 @@ def test_union_categoricals_ordered(self):
         with tm.assertRaisesRegexp(TypeError, msg):
             union_categoricals([c1, c2])
 
+    def test_union_categoricals_sort(self):
+        # GH 13763
+        c1 = Categorical(['x', 'y', 'z'])
+        c2 = Categorical(['a', 'b', 'c'])
+        result = union_categoricals([c1, c2], sort_categories=True)
```
Comment (jreback) [inline pandas/tools/tests/test_concat.py]: can you put companion tests with ``sort_categories=False``

File: pandas/types/concat.py [hunk 3]
```diff
@@ -244,41 +245,47 @@ def union_categoricals(to_union):
...
     if all(first.is_dtype_equal(other) for other in to_union[1:]):
-        return Categorical(np.concatenate([c.codes for c in to_union]),
-                           categories=first.categories, ordered=first.ordered,
-                           fastpath=True)
+        # identical categories - fastpath
+        categories = first.categories
+        ordered = first.ordered
+        new_codes = np.concatenate([c.codes for c in to_union])
+
+        if sort_categories and not categories.is_monotonic_increasing:
```
Comment (sinhrks) [inline pandas/types/concat.py]: do not sort `ordered` categories. needs tests also.

**Comment by jreback** [comment]
lgtm. @sinhrks pls have a look.

**Comment by sinhrks** [comment]
lgtm. pls fix the lint issue:)

**Comment by chris-b1** [comment]
@sinhrks - updated, thanks!

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**